app/models/course/assessment.rb

```diff
@@ -7,9 +7,13 @@ class Course::Assessment < ActiveRecord::Base
   acts_as_conditional
   has_one_folder
 
+  after_initialize :set_defaults, if: :new_record?
+
   before_validation :assign_folder_attributes
   before_validation :propagate_course
 
+  validates :questions, presence: true, unless: :draft?
```

Comment:
does this really check correctly? I don't remember if presence works for associations.